Updating mapField CSE to Use v2 Schema

garetvoit opened the pull request:
- For each version of the component, a corresponding folder labeled v1, v2, ... vX exists. This allows for traditional rule evolution. See [the docs](https://docs.google.com/document/d/1QWYZUWb-gG1I5AKeRF6saKUaLwUk2mIndQ6zkGZaU4o/edit) for more info.

benjamin-kaiser commented:
can we add a description and a better commit message?

john-irby commented on Certified SAIL Extension (CSE) Examples/mapField/appian-extension.xml:
```diff
@@ -6,8 +6,22 @@
+    </parameter>
+    <html-entry-point>index.html</html-entry-point>
+  </component>
+  <component ruleName="mapField" version="2.0.0">
```
I don't think we need to include an example of using the CSE version here - both `mapField` version 1.0.0 & version 2.0.0 are identical

john-irby commented on Certified SAIL Extension (CSE) Examples/mapField/mapField/v2/mapField_en_US.properties:
```diff
@@ -0,0 +1,6 @@
+name=[v2]Map
```
Same comment as above, the v2 example here seems like it may be more confusing for an example reference CSE

garetvoit replied:
@john-irby I'm just using our latest known good example from our test harness. I do think it's valuable to have an example which demonstrates CSE evolution. Perhaps it's worthwhile to split out into a separate example and keep mapField as the basic example?

john-irby replied:
I think making two separate examples with mapField as a base would be best - we could also update version 2.0.0 of the mapField CSE to have noticeably different behavior which would better illustrate the evolution example (though that should probably be split out from this PR)

garetvoit replied:
Removed 2.0.0 in favor of maintaining a bare bones example to get people started with the concepts of CSEs.

garetvoit commented:
@benjamin-kaiser I'm not sure maintaining a separate README is worthwhile here. It introduces another point of failure in terms of forgetting to update documentation. We already maintain in-depth documentation which is linked from the community site.

Updated parent folder name to distinguish between extension and component.

Added more descriptive commit message and description.
